infra/cifuzz/cifuzz.py

```diff
@@ -0,0 +1,149 @@
+    project_name: The name of the OSS-Fuzz project being built.
+    project_repo_name: The name of the projects repo.
+    commit_sha: The commit SHA to be checked out and fuzzed.
+    git_workspace: The location in the shared volume to store git repos.
+    out_dir: The location in the shared volume to store output artifacts.
+
+  Returns:
+    True if build succeeded or False on failure.
+  """
+  if not os.path.exists(git_workspace) or not os.path.exists(out_dir):
+    logging.error('Invalid workspace or out directory.')
+    return False
+
+  src = utils.get_env_var(project_name, 'SRC')
+  if not src:
+    logging.error('Could not get $SRC from project docker image. ')
+    return False
+
+  inferred_url, oss_fuzz_repo_name = build_specified_commit.detect_main_repo(
+      project_name, repo_name=project_repo_name, src_dir=src)
+  if not inferred_url or not oss_fuzz_repo_name:
+    logging.error('Could not detect repo from project %s.', project_name)
+    return False
+
+  # Checkout projects repo in the shared volume.
+  build_repo_manager = repo_manager.RepoManager(inferred_url,
+                                                git_workspace,
+                                                repo_name=oss_fuzz_repo_name)
+  try:
+    build_repo_manager.checkout_commit(commit_sha)
+  except repo_manager.RepoManagerError:
+    logging.error('Error: Specified commit does not exist.')
+    # NOTE: remove return statement for testing.
+    return False
+
+  command = [
+      '--cap-add', 'SYS_PTRACE', '-e', 'FUZZING_ENGINE=libfuzzer', '-e',
+      'SANITIZER=address', '-e', 'ARCHITECTURE=x86_64'
+  ]
+  container = utils.get_container()
+  if container:
+    command += ['-e', 'OUT=' + out_dir, '--volumes-from', container]
+    bash_command = 'rm -rf {0} && cp -r {1} {2} && compile'.format(
+        os.path.join(src, oss_fuzz_repo_name, '*'),
+        os.path.join(git_workspace, oss_fuzz_repo_name), src)
+  else:
+    command += [
+        '-e', 'OUT=' + '/out', '-v',
+        '%s:%s' % (os.path.join(git_workspace, oss_fuzz_repo_name),
+                   os.path.join(src, oss_fuzz_repo_name)), '-v',
+        '%s:%s' % (out_dir, '/out')
+    ]
+    bash_command = 'compile'
+
+  command.extend([
+      'gcr.io/oss-fuzz/%s' % project_name,
+      '/bin/bash',
+      '-c',
+  ])
+  command.append(bash_command)
+
+  if helper.docker_run(command):
+    logging.error('Error: Building fuzzers failed.')
+    return False
+  return True
+
+
+def run_fuzzers(project_name, fuzz_seconds, out_dir):
+  """Runs a all fuzzers for a specific OSS-Fuzz project.
+
+  Args:
+    project_name: The name of the OSS-Fuzz project being built.
+    fuzz_seconds: The total time allotted for fuzzing.
+    out_dir: The location in the shared volume to store output artifacts.
+
+  Returns:
+    (True if run was successful, True if bug was found False if not).
+  """
+  if not out_dir or not os.path.exists(out_dir):
+    logging.error('Error: Unreachable out_dir argument.')
```

Comment:
It's helpful to include the value of `out_dir` in the error message.

Reply:
Done.